src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerGovernmentOrPublicAuthorityValidator.java

```diff
@@ -0,0 +1,112 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.NatureOfControlType;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.BeneficialOwnerGovernmentOrPublicAuthorityDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.CountryLists;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.DateValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.NatureOfControlValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@Component
+public class BeneficialOwnerGovernmentOrPublicAuthorityValidator {
+
+    public static final String NATURE_OF_CONTROL_FIELDS = "nature_of_control";
+
+    private final AddressDtoValidator addressDtoValidator;
+
+    @Autowired
+    public BeneficialOwnerGovernmentOrPublicAuthorityValidator(AddressDtoValidator addressDtoValidator) {
+        this.addressDtoValidator = addressDtoValidator;
+    }
+    
+    public Errors validate(List<BeneficialOwnerGovernmentOrPublicAuthorityDto> beneficialOwnerGovernmentOrPublicAuthorityDtoList, 
+                           Errors errors, String loggingContext) {
+        for (BeneficialOwnerGovernmentOrPublicAuthorityDto beneficialOwnerGovernmentOrPublicAuthorityDto : beneficialOwnerGovernmentOrPublicAuthorityDtoList) {
+            validateName(beneficialOwnerGovernmentOrPublicAuthorityDto.getName(), errors, loggingContext);
+            validateAddress(BeneficialOwnerGovernmentOrPublicAuthorityDto.PRINCIPAL_ADDRESS_FIELD, beneficialOwnerGovernmentOrPublicAuthorityDto.getPrincipalAddress(), errors, loggingContext);
+
+            boolean sameAddressFlagValid = validateServiceAddressSameAsPrincipalAddress(beneficialOwnerGovernmentOrPublicAuthorityDto.getServiceAddressSameAsPrincipalAddress(), errors, loggingContext);
+            if (sameAddressFlagValid && Boolean.FALSE.equals(beneficialOwnerGovernmentOrPublicAuthorityDto.getServiceAddressSameAsPrincipalAddress())) {
+                validateAddress(BeneficialOwnerGovernmentOrPublicAuthorityDto.SERVICE_ADDRESS_FIELD, beneficialOwnerGovernmentOrPublicAuthorityDto.getServiceAddress(), errors, loggingContext);
+            }
```

Comment:
for the address validation, if the 'same as' flag is set to true, then we should ensure the 'other' address fields are not supplied / empty. The web will do this in the post function, if the user ticks the 'same as' option, then the controller will delete the 'other' address. So we'll have to add checks for this in the api to make sure a different address is not supplied if 'same as' is set to true. 
I'll raise a separate story for this change as it will affect the other validators we've done so far.

Reply:
created https://companieshouse.atlassian.net/browse/ROE-1311
Leave the address validation as is in this PR and we'll do ROE-1311 separately.

Reply:
As discussed, need something similar for the 'on other register' question/radio buttons too @markpit .

Reply:
Ah. See you've added this - https://companieshouse.atlassian.net/browse/ROE-1312. Nice.